modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/construction/ConstructionService.scala

```diff
@@ -3,25 +3,40 @@ package org.tessellation.rosetta.domain.construction
 import cats.data.EitherT
 import cats.effect.Async
 import cats.syntax.applicativeError._
+import cats.syntax.either._
+import cats.syntax.flatMap.toFlatMapOps
 import cats.syntax.functor._
 
-import org.tessellation.rosetta.domain.error.{ConstructionError, InvalidPublicKey}
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.rosetta.domain.error.{ConstructionError, InvalidPublicKey, MalformedTransaction}
 import org.tessellation.rosetta.domain.{AccountIdentifier, RosettaPublicKey}
+import org.tessellation.schema.transaction.Transaction
 import org.tessellation.security.SecurityProvider
+import org.tessellation.security.hash.Hash
+import org.tessellation.security.hex.Hex
 import org.tessellation.security.key.ops._
+import org.tessellation.security.signature.Signed
 
 trait ConstructionService[F[_]] {
   def derive(publicKey: RosettaPublicKey): EitherT[F, ConstructionError, AccountIdentifier]
+  def deserializeTransactionFromHex(hex: Hex): EitherT[F, ConstructionError, Hash]
 }
 
 object ConstructionService {
-  def make[F[_]: Async: SecurityProvider](): ConstructionService[F] = new ConstructionService[F] {
+  def make[F[_]: Async: SecurityProvider: KryoSerializer](): ConstructionService[F] = new ConstructionService[F] {
     def derive(publicKey: RosettaPublicKey): EitherT[F, ConstructionError, AccountIdentifier] =
       publicKey.hexBytes
         .toPublicKeyByEC[F]
         .map(_.toAddress)
         .map(AccountIdentifier(_, None))
         .attemptT
         .leftMap(_ => InvalidPublicKey)
+
+    def deserializeTransactionFromHex(hex: Hex): EitherT[F, ConstructionError, Hash] = KryoSerializer[F]
```

Comment:
I would think about the method name. It deserializes transaction as a part of functionality but produces transaction `Hash` as an output not a transaction itself.

Reply:
@TheMMaciek  In rosetta signed transaction is represented by hash `"signed_transaction": "string"` so the name is compatible with rosetta's convention because they didn't name it `signed_transaction_hash`.

But I agree that `deserializeTransactionHashFromHex` is more accurate.

Reply:
I agree with @TheMMaciek, maybe `getTransactionHash(transaction: Hex)`?

Reply:
or with `newtype` so we can say that it is not any hash but transaction's hash so there is a method `fromHex`

```scala
@newtype
case class TransactionHash(hash: Hash)

object TransactionHash {
  def fromHex[F[_]: KryoSerializer](hex: Hex): Either[ConstructionError, TransactionHash]  = KryoSerializer[F].deserialize(...)
}
```
🤔 

and then

```scala
case req @ POST -> Root / "hash" =>
      req.decodeRosettaWithNetworkValidation[ConstructionHash.Request](appEnvironment, _.networkIdentifier) { hashReq =>
        TransactionHash.fromHex(hashReq.signedTransaction)
          .bimap(_.toRosettaError, ConstructionHash.Response(_, None))
          .asRosettaResponse
          .handleUnknownError
      }
```

but I am not sure if we aren't mixing responsibilities then

Reply:
@kpudlik yes, we leak `ConstructionError` out of construction service 🤔

Reply:
Are we going with the `@newtype` then?

Reply:
IMO just change the method name. Do you agree @kpudlik?

Reply:
@marcinwadon yes, it is the quick win without overcomplicating everything :)

Reply:
Changed.